spec/Section 5 -- Validation.md

```diff
@@ -1853,6 +1976,39 @@ fragment isHouseTrainedFragment on Dog {
 This document is not valid because {queryWithExtraVar} defines an extraneous
 variable.
 
+### All Fragment Variables Used
+
+**Formal Specification**
+
+- For every {fragment} in the document:
+  - Let {variables} be the variables defined by that {fragment}.
+  - Each {variable} in {variables} must be used at least once in the fragment's
+    scope.
```

Comment:
I think `scope` might be poorly defined; e.g. it could be interpretted to include fragments spread within it. Perhaps:
```suggestion
  - Each {variable} in {variables} must be used at least once transitively within the fragment's selection set excluding traversal of named fragment spreads.
```